libs/agno/agno/workflow/v2/workflow.py

```diff
@@ -200,23 +202,48 @@ def _execute(
+                    # Handle both single StepOutput and List[StepOutput] (from Parallel/Loop steps)
+                    if isinstance(step_output, list):
+                        # This is a step that returns multiple outputs (Parallel, Loop, etc.)
+                        for output in step_output:
+                            shared_images.extend(output.images or [])
+                            output_images.extend(output.images or [])
+                            shared_videos.extend(output.videos or [])
+                            output_videos.extend(output.videos or [])
+                            shared_audio.extend(output.audio or [])
+                            output_audio.extend(output.audio or [])
+
+                        # Use the last output's content as previous content for chaining
+                        if step_output:
+                            previous_step_content = step_output[-1].content
+
+                        collected_step_outputs.append(step_output)
+                    else:
+                        # This is a regular single step
+                        previous_step_content = step_output.content
+                        shared_images.extend(step_output.images or [])
+                        output_images.extend(step_output.images or [])
+                        shared_videos.extend(step_output.videos or [])
+                        output_videos.extend(step_output.videos or [])
+                        shared_audio.extend(step_output.audio or [])
+                        output_audio.extend(step_output.audio or [])
 
-                    previous_step_content = step_output.content
-                    shared_images.extend(step_output.images or [])
-                    output_images.extend(step_output.images or [])
-                    shared_videos.extend(step_output.videos or [])
-                    output_videos.extend(step_output.videos or [])
-                    shared_audio.extend(step_output.audio or [])
-                    output_audio.extend(step_output.audio or [])
+                        collected_step_outputs.append(step_output)
 
-                    # Collect the StepOutput for storage
-                    collected_step_outputs.append(step_output)
+                    self._collect_workflow_session_state_from_agents_and_teams()
 
                 # Update the workflow_run_response with completion data
-                workflow_run_response.content = collected_step_outputs[
-                    -1
-                ].content  # Final workflow response output is the last step's output
+                if collected_step_outputs:
+                    last_output = collected_step_outputs[-1]
+                    if isinstance(last_output, list) and last_output:
+                        # If it's a list (from Parallel/Loop/etc.), use the last one
+                        workflow_run_response.content = last_output[-1].content
```

Comment:
Hmm this is weird, but I don't have a better solution. 
Probably we will recommend to users to not end with a loop/parallel, but end with a "aggregation" step. But if they don't, then we need to append all the content for the workflow_run_response.content?